### PR TITLE
refactor: replace deprecated substr usages

### LINKED ISSUE #3402: replace String.prototype.substr() with String.prototype.substring()
### Information

[String.prototype.substr()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/substr) is deprecated and should be replaced with [String.prototype.substring()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/substring).

This issue is to find all occurrences of `substr` and replace it with `substring`.
Similar to #3396.

If you want to pick this up, drop a comment below.

### Describe any alternatives/workarounds you're currently using

_No response_

### I would be willing to submit a PR to fix this issue

- [ ] Yes
- [ ] No

### PR DESCRIPTION
**Pull request by billyjov**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] The commit message follows our guidelines: https://github.com/ngrx/platform/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Documentation has been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[x] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?

Closes #3402

## What is the new behavior?

 `substring`  was used instead of `substr`

## Does this PR introduce a breaking change?

```
[ ] Yes
[x] No
```

### CODE REVIEW

**Review by brandonroberts — APPROVED** [review]

**Comment by jamesurobertson** [comment]
String.prototype.substring is not a direct replacement of  String.prototype.substr as this PR suggests with the changes made.

The substring() method returns the part of the string between the start and end indexes, or to the end of the string.
The substr() method returns a portion of the string, starting at the specified index and extending for a given number of characters afterwards.

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/substring
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/substr

File: modules/component-store/schematics-core/utility/project.ts [hunk 1]
```diff
@@ -30,8 +30,8 @@ export function getProjectPath(
 ) {
   const project = getProject(host, options);
 
-  if (project.root.substr(-1) === '/') {
-    project.root = project.root.substr(0, project.root.length - 1);
+  if (project.root.substring(-1) === '/') {
```
Comment (markostanimirovic) [inline modules/component-store/schematics-core/utility/project.ts:33]: `substring` will not work with negative index.

```suggestion
  if (project.root.slice(-1) === '/') {
```

 The same should be applied to other places where `string.substr(-1)` is used.

**Review by markostanimirovic — CHANGES_REQUESTED** [review]

File: projects/ngrx.io/tools/example-zipper/exampleZipper.js [hunk 2]
```diff
@@ -144,8 +144,8 @@ class ExampleZipper {
 
     let gpaths = json.files.map((fileName) => {
       fileName = fileName.trim();
-      if (fileName.substr(0, 1) === '!') {
-        return '!' + path.join(exampleDirName, fileName.substr(1));
+      if (fileName.substring(0, 1) === '!') {
```
Comment (cedricduffournet) [inline projects/ngrx.io/tools/example-zipper/exampleZipper.js:147]: this could be simplify ?
```suggestion
      if (fileName[0] === '!') {
```

Comment (billyjov) [reply projects/ngrx.io/tools/example-zipper/exampleZipper.js:147]: Yes sure. Good catch Thanks

**Review by markostanimirovic — APPROVED** [review]

**Comment by markostanimirovic** [comment]
Thanks @billyjov !

**Pull request merged**